### DISCUSSION TITLE
How extended config overrides my rules

### DISCUSSION THREAD

**Discussion by mattchannn** [eslint/eslint#15753] [Q&A] [answered]
``{
  "env": {
    "browser": true,
    "es2020": true,
    "node": true,
    "es6": true,
    "worker": true
  },
  "extends": ["eslint:recommended", "airbnb-base"],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "rules": {
    "no-unused-vars": "warn",
    "class-methods-use-this": "warn"
  }
}
``

Assume I have above `.eslintrc.json`, what would be the impact if `airbnb-base` turned off some rules but the `eslint:recommended` turns on by default. Which comes first? I would also like to understand more about how configs, plugins, rules work together, like.. which overrides which, because I have no idea after reading the eslint documentation.

**Comment by aladdin-add** [comment, marked as answer]
> what would be the impact if airbnb-base turned off some rules but the eslint:recommended turns on by default.

the rule will be turned off - it's just working like `Object.assign()`. and your own configured rules will overwrite the extended. an example:

```js
{
  "extends": ["eslint:recommended"], // it enabled no-debugger
  "rules": {"no-debugger": "off"}
}
```
so the rule `no-debugger` will be turned off.

**Reply by mattchannn** [reply]
Therefore, the latter config file extended will override the former one, the rules options will also overrides the extended config files.

How about plugins? Actually I am seeking for something like the priority table in the official documentation.

**Reply by aladdin-add** [reply]
plugin is a different thing, it tells eslint to load 3rd-party rules, I just don't  think it has a priority.
the  docs is here: https://eslint.org/docs/developer-guide/shareable-configs. I do agree we can add something about the extending behavior. :)

**Reply by mattchannn** [reply]
> plugin is a different thing, it tells eslint to load 3rd-party rules, I just don't think it has a priority. the docs is here: [eslint.org/docs/developer-guide/shareable-configs](https://eslint.org/docs/developer-guide/shareable-configs). I do agree we can add something about the extending behavior. :)

Thanks for answering my question.
But do you think there is a scenario a 3rd party plugin has some pre-defined rules which overlaps the one defined in config file?
